controllers/authpolicy_status.go

```diff
@@ -82,10 +83,19 @@ func (r *AuthPolicyReconciler) acceptedCondition(policy common.KuadrantPolicy, s
 	return common.AcceptedCondition(policy, specErr)
 }
 
-// enforcedCondition checks if the provided AuthPolicy is enforced based on the status of the associated AuthConfig and Gateway.
+// enforcedCondition checks if the provided AuthPolicy is enforced, ensuring it is properly configured and applied based
+// on the status of the associated AuthConfig and Gateway.
 func (r *AuthPolicyReconciler) enforcedCondition(ctx context.Context, policy *api.AuthPolicy, targetNetworkObject client.Object) *metav1.Condition {
 	logger, _ := logr.FromContext(ctx)
-	authConfigReady, gwPolicyOverridden, err := r.checkAuthConfigAndGateway(ctx, policy)
+
+	// Check if the policy is overridden
+	if r.isPolicyOverridden(policy) {
```

Comment:
This assumes that computing the desired AuthConfig (occasionally none), then marking the AuthPolicy as overridden and finally calculating the Enforced condition happen in a synchronous way, which is a fair assumption because it's the logical path. However, a simple goroutine added somewhere along this flow could break this logic, which spreads across multiple functions. So maybe add a comment here explaining how critical it is to keep this synchronous? WDYT?